src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -5,32 +5,109 @@ import userEvent from "@testing-library/user-event";
   document.body.appendChild(element);
 
-  return renderHook(() =>
-    useActiveDescendantListFocus({
-      focusedElementRef: {
-        current: element
-      },
-      itemsIds: ITEM_IDS,
-      isItemSelectable: isItemSelectable,
-      onItemClick,
-      isHorizontalList: isHorizontal
-    })
-  );
+  const props = {
+    focusedElementRef: {
+      current: element
+    },
+    keepOptionSelected,
+    itemsIds: ITEM_IDS,
+    isItemSelectable: isItemSelectable,
+    onItemClick,
+    isHorizontalList: isHorizontal
+  };
+  return renderHook(argprops => useActiveDescendantListFocus({ ...props, ...argprops }));
 }
 
-function runListUnitTest(isHorizontal) {
+function runListUnitTest({ isHorizontal, keepOptionSelected }) {
   const moveForwardKey = isHorizontal ? "{arrowRight}" : "{arrowDown}";
   const oppositeMoveForwardKey = !isHorizontal ? "{arrowRight}" : "{arrowDown}";
+  const naturalIndex = keepOptionSelected ? 0 : undefined;
+  const naturalIndexAfterInitial = keepOptionSelected ? 0 : -1;
+
+  it("should focus same item after item changed", async () => {
+    const onItemClick = jest.fn();
+    const { result, rerender } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+
+    const beforeIndex = result.current.visualFocusItemIndex;
+    const beforeId = result.current.visualFocusItemId;
+
+    const removeItems = 2;
+    const itemsAfterRemove = ITEM_IDS.slice(removeItems);
+    rerender({ itemsIds: itemsAfterRemove });
+
+    if (!keepOptionSelected) {
+      expect(result.current.visualFocusItemId).toEqual(itemsAfterRemove[beforeIndex]);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex);
+    } else {
+      expect(result.current.visualFocusItemId).toEqual(beforeId);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex - removeItems);
+    }
+  });
+
+  it("should focus correct item on focus", async () => {
+    const onItemClick = jest.fn();
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    expect(result.current.visualFocusItemIndex).toEqual(naturalIndex);
+  });
+
+  it("should focus correct item keyboard forward", async () => {
+    const onItemClick = jest.fn();
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    const keyboardTimes = 2;
```

Comment:
one press is enough

Reply:
wanted two since for one it starts with -1 and for the other 0, so wanted to make sure it's not some value by coincidence

Reply:
the steps count is not matter because it will always be previous + 1 ... but fine it's not critical

Reply:
Yes at the begging it wasnt previous + 2, it was just a number before the changes. 
I still prefer 2 😦